### ISSUE TITLE
[coding-standards] New sniff for `@var` annotations

### ISSUE THREAD

**Issue by PetrHeinz** [shopsys/shopsys#555]
### What is happening

Historically, we have annotated variable types NetBeans-style, like this:
```php
$user = $this->getUser();
/* @var $user \Shopsys\FrameworkBundle\Model\Customer\User */
```

This notation is not PhpDoc complaint, even though PhpDoc is a standard for PHP code and we use it for annotations of classes, methods and properties. This leads to problems with static analysis tooling (PHPStan, specifically) and maybe some IDEs.

In #524 (more specifically in d02483bb750f43ee89efc64e27874279ab0cc1a9), a lot of such occurrences were fixed.

### Expected result

We should annotate variabvles in PhpDoc style, like this:
```php
/** @var \Shopsys\FrameworkBundle\Model\Customer\User $user */
$user = $this->getUser();
```
There should be an automated checker (or, preferably, a fixer) that checks the style of variable annotations. They must follow the structure `/** @var <type> <variable> */` and be directly above an assignment statement.

Maybe there already exists such a checker/fixer.

**Comment by simara-svatopluk** [comment]
What about banning inline annotation?

I would strongly recommend annotated methods instead.

**Comment by PetrHeinz** [comment]
@simara-svatopluk This is not possible in all cases, for example when getting a service from DI container or getting an entity from PersistantReferenceFacade...

**Comment by simara-svatopluk** [comment]
@PetrHeinz You can create annotated private method

**Comment by PetrHeinz** [comment]
@simara-svatopluk I'd rather have inline annotations, but that's matter of personal taste, I guess...

**Comment by simara-svatopluk** [comment]
@PetrHeinz After some thinking about this topic, I find these inline annotations as a sign of bad code. And also my suggestion of private annotate methods is bad too.

It is bad practice because it uses service locator - `ControllerTrait` (oh... trait?!) uses `$this->`**`container`**`->get()->getToken()->getUser()`
And it is also bad practice because it brakes Law of Demeter.

Instead I suggest to use proper dependency injection
:heavy_check_mark: no need for inline annotations
:heavy_check_mark: clean code

**Comment by vitek-rostislav** [comment]
This might be important once we want to enable PHPStan lvl 4 as the "netbeans" style is reported as a violation in such a case.

I believe there is some fixer for that that we can use.

**Comment by s3tezsky** [comment]
fixed in #1643
